Blocklist certain modules for weights_only load

mikaylagawarecki opened the pull request:
Also bold certain text in the error message as suggested 
<img width="3000" alt="Screenshot 2024-07-19 at 5 56 48 PM" src="https://github.com/user-attachments/assets/378f20c5-c6b2-4e53-8eaf-0bd26c3a6b60">

With a GLOBAL like `os.execv` the error message is now as such

```python
File "/data/users/mg1998/pytorch/torch/serialization.py", line 1256, in load
    raise pickle.UnpicklingError(_get_wo_message(str(e))) from None
_pickle.UnpicklingError: Weights only load failed. Re-running `torch.load` with `weights_only` set to `False` will likely succeed, but it can result in arbitrary code execution. Do it only if you got the file from a trusted source.
Trying to load unsupported GLOBAL posix.execv whose module posix is blocked.

Check the documentation of torch.load to learn more about types accepted by default with weights_only https://pytorch.org/docs/stable/generated/torch.load.html.
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131259

mikaylagawarecki committed:
Update

[ghstack-poisoned]

mikaylagawarecki committed:
Update

[ghstack-poisoned]

malfet commented on torch/_weights_only_unpickler.py:
```diff
@@ -72,6 +72,14 @@
 
+# modules in this list are never allowed, even if the user attempts to allowlist
+# functions/classes from them
+_blacklisted_modules = [
```
```suggestion
_blocklisted_modules = [
```

malfet commented on torch/_weights_only_unpickler.py:
```diff
@@ -221,6 +229,10 @@ def load(self):
                     elif module in IMPORT_MAPPING:
                         module = IMPORT_MAPPING[module]
                 full_path = f"{module}.{name}"
+                if module in _blacklisted_modules:
```
```suggestion
                if module in _blocklisted_modules:
```

malfet commented on torch/_weights_only_unpickler.py:
```diff
@@ -221,6 +229,10 @@ def load(self):
                 full_path = f"{module}.{name}"
+                if module in _blacklisted_modules:
+                    raise RuntimeError(
+                        f"Trying to load unsupported GLOBAL {full_path} whose module {module} is blacklisted."
```
```suggestion
                        f"Trying to load unsupported GLOBAL {full_path} whose module {module} is blocked."
```

mikaylagawarecki committed:
Update

[ghstack-poisoned]

mikaylagawarecki committed:
Update

[ghstack-poisoned]

mikaylagawarecki committed:
Update

[ghstack-poisoned]

mikaylagawarecki committed:
Update

[ghstack-poisoned]

mikaylagawarecki committed:
Update

[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

albanD reviewed (approved):
Thanks

mikaylagawarecki committed:
Update on "Blocklist certain modules for weights_only load"

Also bold certain text in the error message as suggested 
<img width="3000" alt="Screenshot 2024-07-19 at 5 56 48 PM" src="https://github.com/user-attachments/assets/378f20c5-c6b2-4e53-8eaf-0bd26c3a6b60">

With a GLOBAL like `os.execv` the error message is now as such

```python
File "/data/users/mg1998/pytorch/torch/serialization.py", line 1256, in load
    raise pickle.UnpicklingError(_get_wo_message(str(e))) from None
_pickle.UnpicklingError: Weights only load failed. Re-running `torch.load` with `weights_only` set to `False` will likely succeed, but it can result in arbitrary code execution. Do it only if you got the file from a trusted source.
Trying to load unsupported GLOBAL posix.execv whose module posix is blocked.

Check the documentation of torch.load to learn more about types accepted by default with weights_only https://pytorch.org/docs/stable/generated/torch.load.html.
```




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki committed:
Update on "Blocklist certain modules for weights_only load"

Also bold certain text in the error message as suggested 
<img width="3000" alt="Screenshot 2024-07-19 at 5 56 48 PM" src="https://github.com/user-attachments/assets/378f20c5-c6b2-4e53-8eaf-0bd26c3a6b60">

With a GLOBAL like `os.execv` the error message is now as such

```python
File "/data/users/mg1998/pytorch/torch/serialization.py", line 1256, in load
    raise pickle.UnpicklingError(_get_wo_message(str(e))) from None
_pickle.UnpicklingError: Weights only load failed. Re-running `torch.load` with `weights_only` set to `False` will likely succeed, but it can result in arbitrary code execution. Do it only if you got the file from a trusted source.
Trying to load unsupported GLOBAL posix.execv whose module posix is blocked.

Check the documentation of torch.load to learn more about types accepted by default with weights_only https://pytorch.org/docs/stable/generated/torch.load.html.
```




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge
